app/src/main/java/org/elasticsearch/node/internal/CrateSettingsPreparer.java

```diff
@@ -96,24 +106,59 @@ static void validateKnownSettings(Settings.Builder settings) {
         }
     }
 
-    private static void applyCrateDefaults(Settings.Builder settingsBuilder) {
+    @VisibleForTesting
+    static void applyCrateDefaults(Settings.Builder settingsBuilder) {
         // read also from crate.yml by default if no other config path has been set
         // if there is also a elasticsearch.yml file this file will be read first and the settings in crate.yml
         // will overwrite them.
         putIfAbsent(settingsBuilder, TRANSPORT_TYPE_DEFAULT_KEY, Netty3Plugin.NETTY_TRANSPORT_NAME);
         putIfAbsent(settingsBuilder, SETTING_HTTP_PORT.getKey(), Constants.HTTP_PORT_RANGE);
         putIfAbsent(settingsBuilder, PORT.getKey(), Constants.TRANSPORT_PORT_RANGE);
-        putIfAbsent(settingsBuilder, "network.host", "0.0.0.0");
+        putIfAbsent(settingsBuilder, NetworkService.GLOBAL_NETWORK_HOST_SETTING.getKey(), "0.0.0.0");
 
         // Set the default cluster name if not explicitly defined
         if (settingsBuilder.get(ClusterName.CLUSTER_NAME_SETTING.getKey()).equals(ClusterName.DEFAULT.value())) {
             settingsBuilder.put(ClusterName.CLUSTER_NAME_SETTING.getKey(), "crate");
         }
+
+        // Set a random node name if none is explicitly defined
+        if (settingsBuilder.get(Node.NODE_NAME_SETTING.getKey()) == null) {
+            settingsBuilder.put(Node.NODE_NAME_SETTING.getKey(), randomNodeName());
+        }
     }
 
     private static <T> void putIfAbsent(Settings.Builder settingsBuilder, String setting, T value) {
         if (settingsBuilder.get(setting) == null) {
             settingsBuilder.put(setting, value);
         }
     }
+
+    private static String randomNodeName() {
+        List<String> names = nodeNames();
+        int index = ThreadLocalRandom.current().nextInt(names.size());
```

Comment:
Could be changed to pass the index into `nodeNames` so it becomes unnecessary to create a list.